src/Yasumi/Yasumi.php

```diff
@@ -186,7 +189,7 @@ public static function createByISO3166_2($iso3166_2, $year = null, $locale = sel
         $availableProviders = self::getProviders();
 
         if (false === isset($availableProviders[$iso3166_2])) {
-            throw new InvalidArgumentException(sprintf(
+            throw new UnknownIso31662Exception(sprintf(
```

Comment:
I believe `UnknownIso31662Exception` should be removed completely, because function is trying to create an instance of class by locale, so the `ProviderNotFoundException` is more accurate.

Reply:
Yep I totally agree with that ... the error is thrown when no provider has been found for a given isoCode :+1:

@qneyrat please throw a `ProviderNotFoundException` here